lib/solidus_sitemap/engine.rb

```diff
@@ -1,26 +1,25 @@
+# frozen_string_literal: true
+
 module SolidusSitemap
   class Engine < Rails::Engine
     require 'spree/core'
     isolate_namespace Spree
     engine_name 'solidus_sitemap'
 
-    config.autoload_paths += %W(#{config.root}/lib)
+    config.autoload_paths += %W[#{config.root}/lib]
 
-    def self.activate
-      Spree::Product.class_eval do
-        def self.last_updated
-          last_update = order('spree_products.updated_at DESC').first
-          last_update.try(:updated_at)
-        end
+    config.to_prepare do
+      Dir.glob(File.join(File.dirname(__FILE__), '../../app/**/*_decorator*.rb')) do |c|
+        Rails.configuration.cache_classes ? require(c) : load(c)
       end
 
       require 'solidus_sitemap/solidus_defaults'
+
       SitemapGenerator::Interpreter.send :include, SolidusSitemap::SolidusDefaults
-      if defined? SitemapGenerator::LinkSet
-        SitemapGenerator::LinkSet.send :include, SolidusSitemap::SolidusDefaults
-      end
-    end
 
-    config.to_prepare(&method(:activate).to_proc)
+      return unless defined? SitemapGenerator::LinkSet
+
+      SitemapGenerator::LinkSet.send :include, SolidusSitemap::SolidusDefaults
```

Comment:
I think I would keep the `unless` block here, because that way you can easily add instructions after this one that don't depend on the condition. I know Rubocop complains about it, but usually I just disable that rule because it doesn't make much sense.